src/types.ts

```diff
@@ -49,7 +49,23 @@ export type ZodTypeAny = ZodType<any, any, any>;
 export type TypeOf<T extends ZodType<any, any, any>> = T["_output"];
 export type input<T extends ZodType<any, any, any>> = T["_input"];
 export type output<T extends ZodType<any, any, any>> = T["_output"];
-export type { TypeOf as infer };
+export type TypeOfFlattenedError<
+  T extends ZodType<any, any, any>,
+  U = string
+> = {
+  formErrors: U[];
+  fieldErrors: {
+    [P in keyof TypeOf<T>]?: U[];
+  };
+};
+export type TypeOfFormErrors<
```

Comment:
This could be a single `z.inferFormErrors` type that just has the signature of `z.inferFlattenedError`, but I wanted to avoid confusion between results of `flatten()` vs `formErrors` getter.

What would you prefer? A single type, or is the convince alias type useful?

Reply:
I personally appreciate the alias here since it actually gives a pretty nice hint at what is actually happening.